Dockerfile

```diff
@@ -19,7 +19,14 @@ COPY README.md packages/webapps.e/telepor[t]/package.json web-apps/packages/weba
 
 # download and install npm dependencies
 WORKDIR web-apps
-RUN yarn install
+# Install JavaScript dependencies and manually check if yarn.lock needs an update.
+# Yarn v1 doesn't respect the --frozen-lockfile flag when using workspaces.
+# https://github.com/yarnpkg/yarn/issues/4098
+RUN sha384sum yarn.lock > yarn-lock-sha \
+  && yarn install \
+  && sha384sum --check yarn-lock-sha || \
+  { echo "yarn.lock needs an update; run yarn install, verify that correct dependencies were installed \
+and commit the updated version of yarn.lock"; exit 1; }
```

Comment:
Using a single RUN so that Docker creates only a single intermediate layer.

I used to have this on separate lines but Docker would first take about 20 seconds to update the cache after `RUN yarn install` and only then it'd fail.

Reply:
I like the solution. Just out of curiosity, what made you choose `sha384sum`?

Reply:
I wasn't sure if my knowledge about which SHAs are safe to use was up to date, so I went to https://en.wikipedia.org/wiki/Secure_Hash_Algorithms and picked the lowest one with both green fields. 😶‍🌫️

Though it seems that 256 is okay for checksums, we use 256 for our downloads. And in this scenario probably even md5 would be good enough.

Reply:
Yeah that's fair, I doubt we'd end up with a collision between yarn.lock changes, even if the hashing algorithm has been proved to collide (md5 for example)